### ISSUE TITLE
PDF generation failed. Cannot find module 'C:\Users\Username\.atom\packages\fountain\configs\afterwritingFonts.json.js'

### ISSUE THREAD

**Issue by trochitv** [superlou/fountain-atom#110]
Hi,

after I resolved my problems of not being able to install fountain by installing git, I now face the issue of not being able to generate pdfs from my .fountain files.

I installed the afterwriting cli via npm, but when I try to render the preview I am faced with this error:

![image](https://user-images.githubusercontent.com/59615400/100520897-b66c2080-31a0-11eb-998c-2957a5567956.png)

and no pdf.

Thank you for your time.

**Comment by trochitv** [comment]
Anyone?

**Comment by superlou** [comment]
@trochitv I don't think you should've needed to install Afterwriting manually, though I don't think that should've hurt anything. If you aren't using the Afterwriting CLI, can you try uninstalling that, uninstalling the Fountain-Atom package, then reinstalling the Fountain-Atom package?

On Linux Mint, I'm not getting the error you are, but it is saying "An error occurred while loading the PDF," and I don't see a PDF file being created. Do you get that message?

**Comment by jrbostic** [comment]
It's extremely odd that it would be looking for a file with the extension ".json.js".  The json file is legit part of rendering with afterwriting... But what is requesting that file with js extension, I have no idea.  It's definitely wrong though.

**Comment by trochitv** [comment]
Thank you so much for the answers!
It didn't show me a pdf when I had not installed afterwriting yet, so I went ahead and did. Pretty sure it was the very same issue, but I will try with it uninstalled later.

It says "pdf file generated" but apparently it never is. I do not get an explicit error concerning the pdf.

Is there anything else I can try or more info I can give you?

**Comment by superlou** [comment]
For sanity, can you check if Fountain outline view is working? That will at
least confirm if the package thinks it's mostly installed.

**Comment by trochitv** [comment]
![image](https://user-images.githubusercontent.com/59615400/101262316-fcdff300-373d-11eb-96eb-781994ba4ec4.png)
This is what that looks like.

**Comment by superlou** [comment]
If you click one more time in the editing pane, e.g. on EDWARD, does the outline finish populating? Something looks very weird.

**Comment by trochitv** [comment]
Nope. "INT. WILL'S BEDROOM -NIGHT (1973)" is all I ever get. :/

**Comment by trochitv** [comment]
With the entire big fish script it looks like this:
![image](https://user-images.githubusercontent.com/59615400/101263171-978f0080-3743-11eb-9c46-fd064d4dca35.png)

**Comment by superlou** [comment]
I might have something. If I update to the latest Afterwriting release, the PDF generation tests start to fail and I receive a similar error:

![image](https://user-images.githubusercontent.com/709695/101284158-98a84800-37ac-11eb-8de0-908b85df1e5a.png)

@trochitv, can you go into the Fountain installation and look in `package-lock.json` for the version of the Afterwriting dependency? @jrbostic Is there any chance Afterwriting has changed the inputs to rendering?

**Comment by superlou** [comment]
Pretty sure that's the problem: https://github.com/ifrost/afterwriting-labs/blob/master/docs/clients.md#custom-fonts.

**Comment by jrbostic** [comment]
@superlou It wouldn't surprise me... But the Afterwriting dependency version in package.json shouldn't have that particular change yet, based on what I saw in the merge comments for that change.

**Comment by superlou** [comment]
Agreed. I'm wondering if something happened when @trochitv installed the Afterwriting CLI. That said, I'm not sure why it would affect the Afterwriting version the atom package is using.

**Comment by superlou** [comment]
@jrbostic I pushed an update to trunk that uses the latest Afterwriting, with some hacks to use the new font loading scheme. The test suite passes, but it doesn't really test that functionality. I'm not sure if overwriting existing PDFs is working, though the tests pass. Can you take a look when you get a chance and see if it works for you?

**Comment by trochitv** [comment]
@superlou it says version 1.12.7

**Comment by trochitv** [comment]
Just recreated it on my Ubuntu machine. Didn't install afterwriting there.

**Comment by superlou** [comment]
@trochitv Can you try running the development version on the trunk?

**Comment by trochitv** [comment]
I'm sorry, where may I find the development version? >.<

**Comment by superlou** [comment]
@trochitv, you should be able to get it by uninstalling the Fountain you have installed, then doing `apm install fountain` in a terminal.

**Comment by johnygum** [comment]
Hello! I have the same issue. I am running Debian 10. I have installed the package through the Atom UI and also with `apm install fountain` and nothing worked. Thanks for the help!

When I open Developer tools in Atom, I get basically the same error as in this screenshot:
> ![image](https://user-images.githubusercontent.com/709695/101284158-98a84800-37ac-11eb-8de0-908b85df1e5a.png)

**Comment by superlou** [comment]
Sorry, I typo'd in my earlier comment. Please uninstall the package again, then try `apm develop fountain`.

**Comment by johnygum** [comment]
Hi, Thanks, this kind of worked, with `apm develop fountain`, cloned github repo into /home/user/github/fountain and made a link to /home/user/.atom/dev/packages/fountain, but when I restarted Atom, the package didn't work, I checked in the package list and saw Fountain under development packages, but it said, it wasnt installed, there was only an "install" button. I clicked that button, and then it installed the old version, that didn't work... I then just overriden everything from /home/user/.atom/packages/fountain with  /home/user/github/fountain and now it works. I am pretty sure this is not the correct way to install dev version... I am not very familiar with Atom packages... Thanks again

**Comment by superlou** [comment]
@johnygum That's probably not a great way, but I think that successfully got the latest source installed as a plugin. 😄 I think `apm install git@github.com:superlou/fountain-atom.git` might also have worked, but I'm not sure. Can you summarize what wasn't working and double check that it is now working?

In theory, if everything is working, we can publish this source as a release so it's easier to install. However, it's a pretty hacky change to some of @jrbostic 's code, so I was hoping he would weigh in.

**Comment by johnygum** [comment]
Hi, i uninstalled the package and installed it with `apm install https://github.com/superlou/fountain-atom.git` and now it is working fine.
Before, I installed the package through Atom UI, and it installed fine, and outline view worked, but pdf generation and preview didn't.
I noticed a problem when I try to generate a PDF with `Ctrl + Alt + Shift + X` when the pdf already exists, it doesn't overwrite it, even though I selected overwrite when Atom opens the message. I see no error in the console window. If I delete the pdf file and generate it again, it works fine. I am not sure if these problems are related.

**Comment by superlou** [comment]
@johnygum I can confirm the same problem with PDFs not being overwritten. The weird thing is we have a test case that is supposed to check for that which is still passing. I don't think anything about the latest Afterwriting should affect overwriting PDFs, but that does seem like the most likely culprit.

**Comment by superlou** [comment]
@johnygum PDF overwriting is tentatively fixed in the latest master. Can you try uninstalling and reinstalling from github?

**Comment by jrbostic** [comment]
@superlou  Sorry, I only saw part of this thread last weekend.  I'm trying to figure out where/how to set this project up on one of my newer machines.  Re-establish some kind of workflow.  Took me a lot of dedicated time a few years ago... Can't remember any of it now... Lol

**Comment by superlou** [comment]
@jrbostic No worries. Every time I come back to this I have to relearn the toolchain, and it doesn't help that all of the docs have switched back to Javascript.

**Comment by johnygum** [comment]
> @johnygum PDF overwriting is tentatively fixed in the latest master. Can you try uninstalling and reinstalling from github?

Hi, I installed the new version and now everything is working!

Thanks very much!

**Comment by trochitv** [comment]
Working for me as well. Thank you very much :)
